test.js

```diff
@@ -54,13 +55,14 @@ test('toVFile()', function (t) {
   })
 
   t.test('should accept a WHATWG URL object', function (st) {
-    var file = toVFile(new URL('file:///foo/bar/baz.qux'))
+    const dir = fileURLToPath(new URL('./', import.meta.url))
+    var file = toVFile(new URL('./baz.qux', import.meta.url))
 
-    st.equal(file.path, join('/foo', 'bar', 'baz.qux'))
+    st.equal(file.path, join(dir, 'baz.qux'))
     st.equal(file.basename, 'baz.qux')
     st.equal(file.stem, 'baz')
     st.equal(file.extname, '.qux')
-    st.equal(file.dirname, join('/foo', 'bar'))
+    st.equal(file.dirname, dir.replace(/[/\\]$/, ''))
```

Comment:
Is this one needed for Windows? I’d guess that Node now has the correct separators in `dir`?

Reply:
It is necessary to remove the trailing slash (or backslash on Windows). That's because `new URL('.', base)` leaves a trailing slash, which is not lost during the translation to a path.

Reply:
ah, ok :+1: